src/server/controllers/movie.js

```diff
@@ -1,31 +1,66 @@
-const jwt = require('jsonwebtoken');
-const { PrismaClient } = require('@prisma/client');
+const jwt = require("jsonwebtoken");
+const { PrismaClient } = require("@prisma/client");
 const prisma = new PrismaClient();
 
-const jwtSecret = 'mysecret';
+const jwtSecret = "mysecret";
 
 const getAllMovies = async (req, res) => {
-    const movies = await prisma.movie.findMany();
+  let payload;
+  try {
+    const [_, token] = req.headers.authorization.split(" ");
+
+    payload = jwt.verify(token, jwtSecret);
+  } catch (e) {
+    return res.status(401).json({ error: "Invalid token provided." });
+  }
+
+  const user = await prisma.user.findFirst({
+    where: { username: payload.username },
```

Comment:
Username probably should be unique to guarantee you don't return a user different to the user identified in the token.

In which case, you should then be able to use prisma's findUnique query

Reply:
Is there a scenario where a user creates a token, then the user is removed from the user table, the JWT is validated, and then prisma tries to find the user to then find movies?

If so, you should reject the request if there is no user will an error message.

Reply:
Now using findUnique to look for users.

Would something like this be sufficient to cover the scenario?

`const foundUser = await prisma.user.findUnique({
    where: { username: payload.username },
  });

  if (!foundUser) {
    return res.status(401).json({ error: "Invalid username." });
  }`